File: code/library/src/main/java/org/fbme/lib/iec61499/instances/FunctionBlockInstance.java
package org.fbme.lib.iec61499.instances;


import org.fbme.lib.iec61499.fbnetwork.FunctionBlockDeclarationBase;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface FunctionBlockInstance extends Instance {

    @NotNull NetworkInstance getParent();

    @NotNull FunctionBlockDeclarationBase getDeclaration();

    @Nullable NetworkInstance getContainedNetwork();
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/instances/FunctionBlockInstance.kt
package org.fbme.lib.iec61499.instances

import org.fbme.lib.iec61499.fbnetwork.FunctionBlockDeclarationBase

interface FunctionBlockInstance : Instance {
    override val parent: NetworkInstance
    override val declaration: FunctionBlockDeclarationBase
    val containedNetwork: Instance?
}
